### PR TITLE
Nonconformity: Remove deprecated type attribute in ir.ui.view + Cleanup and Refactor

### PR DESCRIPTION
**Pull request by bwrsandman**
This is a simple cleanup of this module due while fixing a deprecation
- Remove deprecated `type` attribute in ir.ui.view
-  Refactor module:
  - Separate data, demo from views
  - Separate view, data and python scripts by model
  - Use CamelCase for class names
- Fixup issues with MgmtsystemNonconformity
  - Remove redundant _rec_name
  - Replace Static STATES by function which has access to context for translation and which is inheritable
  - Remove _STATES_DICT and replace with dynamic get of states
- Update translation files

The changes are in 213a699 and 3736bd0
There is only refactoring in fb69405
And 69ef027 is only pot/po files and one small string change.

### CODE REVIEW

File: mgmtsystem_nonconformity/model/mgmtsystem_nonconformity.py [hunk 1]
```diff
@@ -277,12 +153,29 @@ def _state_name(self, cr, uid, ids, name, args, context=None):
...
+            ('analysis', _('Analysis')),
+            ('pending', _('Pending Approval')),
+            ('open', _('In Progress')),
+            ('done', _('Closed')),
+            ('cancel', _('Cancelled')),
+        ]
+
     def create(self, cr, uid, vals, context=None):
         vals.update({
             'ref': self.pool.get('ir.sequence').get(
```
Comment (lmignon) [inline mgmtsystem_nonconformity/model/mgmtsystem_nonconformity.py]:
``` python
self.pool['ir.sequence']
```

**Comment by lmignon** [comment]
:+1:  (Code review)

**Comment by max3903** [comment]
:+1:

**Comment by bwrsandman** [comment]
This should be ready to merge

**Pull request merged**